agents/builtin/mysql/mysql.go

```diff
@@ -49,42 +59,64 @@ type Params struct {
 // Change represents Agent status change _or_ QAN collect request.
 type Change struct {
```

Comment:
I don't know, but I think it's bad when one thing in public API represents more than one meaning. These things are different by nature, isn't it? I mean it has different areas of responsibility. `AgentStatus` is about the agent lifecycle but `CollectRequest` is about collecting data.

I mean It's like some function with the name `AddOrJoin(a, b int) int`.

Reply:
Indeed, there were two channels. But they were harder to use and much easier to misuse. Supervisor reads changes channel until it is closed. If there are two of them, and one is closed – should we wait for the second to close, or not? Should we drain the data channel before removing that Agent? etc etc. Supervisor code become much more complex.
Again, we could revisit it once we have more other Agents.